### DISCUSSION TITLE
How to set nested object to null

### DISCUSSION THREAD

**Discussion by fabiosussetto** [react-hook-form/react-hook-form#3395] [Q&A] [answered]
Hi, first of all thanks for all the hard work that's gone into this library.

I have a question: I'm using react-hook-form to work with a nested object, like this:

```
{
   title: "Testing",
   image: {
        path: "foo.png"
   }
}
```

Using ```Controller``` I can correctly edit the nested image.path attribute by using dot notation in the field name.
However, when the user presses a button I'd like to clear the whole image object, effectively setting the ```image``` key to null. The expected output should at this point be:

```
{
   title: "Testing",
   image: null
}
```

Thing is though, I can't just do ```setValue('image', null)```, I believe because there's no field registered with the name "image" (but there's "image.path" instead. 

Any advice on how I can approach this? Basically, what I'm trying to achieve is similar to how useFieldArray works, except with just an object instead of an array of objects.

Thanks in advance.

**Comment by bluebill1049** [comment]
could you reproduce that in a CSB?

**Reply by eledsoft** [reply]
I have the same problem. I have reproduced it in this codesandbox https://codesandbox.io/s/react-hook-form-v7-controller-forked-e8dxy  Also, I noticed that RHF reports an error when i clicked on field.
How can I resolve the problem?

**Reply by eledsoft** [reply]
I forgot: The problem is on   "items.TextField" field

**Reply by Moshyfawn** [reply]
It doesn't look like the same problem to me.
From what I see, you have a `items.TextField` field that you're trying to update by having a controlled input with name `items.TextField`. But RHF treats field names with dots as a dot notation, so it updates a field `TextField` inside the `items` object. You can see this `items` object has your <TextField /> value in your submitted data output
```json
"items": {
  "TextField": "Text"
}
```
while the `items.TextField` field has it's default value of `null`, since no input is bound to this name.

I'd encourage you to exclude special characters from fields names (like the `.`).

Refer to [RHF docs on register API](https://react-hook-form.com/api/useform/register) for more information on how RHF treats field names. Same logic applies to `<Controller />`'s `name` prop.

**Reply by eledsoft** [reply]
I have corrected according to your indications and if as default value I use any non-null value, everything works.
But if the default value is null, then I get an error when I click in the field and the reset button no longer works.
How can I do to avoid this problem?

**Reply by Moshyfawn** [reply]
I'd suggest to voiding setting the text input's default value as `null`. You can see this warning in the console about controlled vs uncontrolled components when use `null` default value
```bash
Warning: `value` prop on `input` should not be null. Consider using an empty string to clear the component or `undefined` for uncontrolled components.
```

**Comment by Moshyfawn** [comment, marked as answer]
To answer the initial question _(I realize it most probably is very stale by now)_, `setValue` approach looks fine. See [this CSB](https://codesandbox.io/s/nervous-cohen-md1ex?file=/src/App.tsx) for example
